kbs/src/api/src/http/attest.rs

```diff
@@ -17,11 +18,19 @@ pub(crate) async fn auth(
     request: web::Json<Request>,
     map: web::Data<SessionMap>,
     timeout: web::Data<i64>,
+    attestation_service: web::Data<Arc<AttestationService>>,
 ) -> Result<HttpResponse> {
     info!("Auth API called.");
     debug!("Auth Request: {:?}", &request);
 
-    let session = SessionStatus::auth(request.0, **timeout)
+    let extra_params = attestation_service
+        .generate_supplemental_challenge(request.tee, request.extra_params.clone())
```

Comment:
we have #242 RFC to suggest how `extra_params` are used. perhaps we can review the new usages in it first and have them documented before any code gets added.

Reply:
Sounds good. The only flaws not covered well in this PR is how KBS interacts with AS. In the beginning we have refactored AS & KBS to [avoid coupling](https://github.com/confidential-containers/trustee/pull/240). So the core idea is that we should not use a structure of KBS as AS's API.

An alternative way is to put the things into `nonce` rather than `extra_params` for IBM SE here. `extra_params` could be decided & appended in the kbs side by the AS plugin type.


btw, I still prefer to integrate IBM SE inside only KBS part rather than also covering AS part. Let's see in future.

Reply:
In this way, the [nonce()](https://github.com/confidential-containers/trustee/pull/163/commits/1e3b172a5c16d357f68799e0b2ed4c0075fdc362#diff-b83a2a4e4135ca8f4193335a1a8d660b5f1dadf86f8cad301b44c4f2a6867a2eR37) API means the same as `generate_supplemental_challenge()` here.

Reply:
> `generate_supplemental_challenge`

It's similar for `nonce()` and `generate_supplemental_challenge()`, the difference is: for IBM SE which is generated by verifier as part of the challenge. Do you think reasonable to delegate nonce() generation from KBS to verifier?

Reply:
The key point is should AS or KBS generate the attestation challenge. If challenge is specific to verifier or holds verifier specific info, it should be generated by AS/Verifier (IBM SE is such as case).

Reply:
I know that. What I mean are
1. The challenges generated by IBM SE might better to be put to `nonce` rather than `extra_params`
2. `nonce` in @mythi 's PR also aims to be from AS (although the default impl is from KBS), thus the same as `generate_supplemental_challenge`
3. `extra_params` would have some other semantice as #242 proposes. All the contents now is determined by the AS type, but no need to interact with AS to get.
4. Also, as 1stR-A in R-C-A-R are both handled by backend AS, we could rename `generate_supplemental_challenge` in KBS side to `generate_challenge` and directly returns a `Challenge` structure.

Let me write a commit for this.

Reply:
The original names is `generate_challenge` :-)

Reply:
See this commit https://github.com/Xynnn007/kbs/commit/1655c266f67cbdb421659366af655e50662cb68f

Reply:
> See this commit [Xynnn007@1655c26](https://github.com/Xynnn007/kbs/commit/1655c266f67cbdb421659366af655e50662cb68f)

I agree with this approach, this is actually my original thoughts :-), @Xynnn007 do you want me picking it and revise or do you want get the commit merged firstly?

Reply:
If possible, pick it together with this pr

Reply:
LGTM too, seems it also allows to set  per-AS `"supported-hash-algorithms": ["sha256", "sha384", "sha512"]` as suggested by #242.

But perhaps squash the commits a  bit?

Reply:
@Xynnn007 I changed the guest-components accordingly also here https://github.com/confidential-containers/guest-components/pull/492/commits/cdeabd46330defd3e401175a1d7b3c4bf7e8e8e3